src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -399,33 +399,53 @@ Status ServiceBasedNodeInfoAccessor::AsyncSubscribeToNodeChange(
     const StatusCallback &done) {
   RAY_LOG(DEBUG) << "Subscribing node change.";
   RAY_CHECK(subscribe != nullptr);
-  ClientTable &client_table = client_impl_->GetRedisGcsClient().client_table();
-  auto status = client_table.SubscribeToNodeChange(subscribe, done);
+  RAY_CHECK(node_change_callback_ == nullptr);
+  node_change_callback_ = subscribe;
+
+  auto on_subscribe = [this](const std::string &id, const std::string &data) {
+    GcsNodeInfo node_info;
+    node_info.ParseFromString(data);
+    HandleNotification(node_info);
+  };
+
+  auto on_done = [this, subscribe, done](const Status &status) {
+    // Get nodes from GCS Service.
+    auto callback = [this, subscribe, done](
+                        const Status &status,
+                        const std::vector<GcsNodeInfo> &node_info_list) {
+      for (auto &node_info : node_info_list) {
+        HandleNotification(node_info);
+      }
+      if (done) {
+        done(status);
+      }
+    };
+    RAY_CHECK_OK(AsyncGetAll(callback));
```

Comment:
Can you update the comment of `AsyncSubscribeToNodeChange` and indicate that the callback will be also be called for existing nodes?

Reply:
fixed